pkg/register/register.go

```diff
@@ -94,17 +78,57 @@ func Register(url string, caCert []byte, smbios bool, emulatedTPM bool, emulated
 	_ = conn.SetWriteDeadline(time.Now().Add(RegistrationDeadlineSeconds * time.Second))
 	_ = conn.SetReadDeadline(time.Now().Add(RegistrationDeadlineSeconds * time.Second))
 
+	logrus.Debug("start TPM attestation")
 	err = tpmAuth.Init(conn)
 	if err != nil {
 		return nil, fmt.Errorf("failed TPM based authentication: %w", err)
 	}
 
-	_, msg, err := conn.NextReader()
+	msgType, _, err := ReadMessage(conn)
 	if err != nil {
-		return nil, fmt.Errorf("reading elemental config: %w", err)
+		return nil, fmt.Errorf("expecting auth reply: %w", err)
 	}
+	if msgType != MsgReady {
+		return nil, fmt.Errorf("expecting '%v' but got '%v'", MsgReady, msgType)
+	}
+	logrus.Info("TPM attestation successful")
 
-	return ioutil.ReadAll(msg)
+	err = sendData(conn, smbios, labels)
+	if err != nil {
+		return nil, err
+	}
+
+	logrus.Debug("get elemental configuration")
+	if err := WriteMessage(conn, MsgGet, []byte{}); err != nil {
+		return nil, fmt.Errorf("request elemental configuration: %w", err)
+	}
+	_, r, err := conn.NextReader()
+	if err != nil {
+		return nil, fmt.Errorf("read elemental configuration: %w", err)
+	}
+	return ioutil.ReadAll(r)
+}
+
+func sendData(conn *websocket.Conn, smbios bool, labels map[string]string) error {
+	if smbios {
+		logrus.Debug("send SMBIOS data")
+		data, err := dmidecode.Decode()
+		if err != nil {
+			return errors.Wrap(err, "failed to read dmidecode data")
+		}
+		err = SendJSONData(conn, MsgSmbios, data)
+		if err != nil {
+			return fmt.Errorf("sending SMBIOS data: %w", err)
```

Comment:
would add debug in here with the data that failed to send in case we find some strange data or broken data (hit: litter.sdump will nicely print it instead of getting a terrible string)

Reply:
thanks, nice advice the linter package 😄 
done! also for the labels... not needed as much as the SMBIOS data (labels are much more under control) but why not? 😁